### DISCUSSION TITLE
Can't get child fetcher to be invoked

### DISCUSSION THREAD

**Discussion by jgsmarques** [Netflix/dgs-framework#192] [Q&A] [answered]
Hi all,

I'm trying to follow the demos on the documentation, and so far I have the same schema as the demo, and a simple class that provides the fetchers:

`schema.graphqls`
```
type Show {
  # The showId may or may not be there, depending on the scenario.
  showId: ID
  title: String
  reviews: [Review]
}

type Review {
  starRating: Int
}
```

`ClientDataFetcher`
```java
@DgsComponent
public class ClientDataFetcher {
  private final List<Show> shows = Arrays.asList(
      new Show("1", "Stranger Things", null),
      new Show("2", "Ozark", null)
  );

  private final List<Review> reviews = Arrays.asList(
      new Review(1),
      new Review(2),
      new Review(3)
  );

  @DgsData(parentType = "Query", field = "shows")
  public List<Show> shows(@InputArgument("titleFilter") String titleFilter) {
    if(titleFilter == null) {
      return shows;
    }

    return shows.stream().filter(s -> s.getTitle().contains(titleFilter)).collect(Collectors.toList());
  }

  @DgsData(parentType = "Show", field = "reviews")
  List<Review> reviews(DgsDataFetchingEnvironment dfe) {
    Show show = dfe.getSource();
    return reviews;
  }
}
```

When running the application, I can make the following query
```
{
  shows {
    showId
    title
    reviews {
      starRating
    }
  }
}
```

and I get the following result
```
{
  "data": {
    "shows": [
      {
        "showId": "1",
        "title": "Stranger Things",
        "reviews": null
      },
      {
        "showId": "2",
        "title": "Ozark",
        "reviews": null
      }
    ]
  }
}
```
 When debugging, the execution never passes through the method `reviews`. Can you please point me in the right direction?
I'm using version 3.8.2 of `graphql-dgs-spring-boot-starter` but I've tried with 3.9.3 with the same result.

Thank you!

**Comment by jgsmarques** [comment, marked as answer]
Just figured it out, I was missing the `public` modifier in the `reviews` method!
Thanks!

**Reply by GrantGochnauer** [reply]
Thank you for this! I ran into the EXACT same problem.

**Reply by jgsmarques** [reply]
This might actually be a bug in the documentation:
https://netflix.github.io/dgs/datafetching/
There are no `public` modifiers in the examples which can be misleading.

**Reply by GrantGochnauer** [reply]
Yes - good catch! And I was following along with the documentation which is why I copied it exactly.

**Reply by GrantGochnauer** [reply]
cc: @paulbakker

**Comment by srinivasankavitha** [comment]
Thanks for pointing this out. Will fix.

**Comment by srinivasankavitha** [comment]
Fixed by https://github.com/Netflix/dgs/pull/27.

**Reply by GrantGochnauer** [reply]
thank you!
